Sofa/Component/IO/Mesh/tests/MeshExporter_test.cpp

```diff
@@ -68,142 +79,181 @@ class MeshExporter_test : public BaseSimulationTest,
-    std::vector<string> params = GetParam() ;
-    ASSERT_EQ(params.size(), NUM_PARAMS );
-    ASSERT_NO_THROW( this->checkBasicBehavior(params, tempdir+"/invalid1/invalid2/invalid3/outfile", {tempdir+"/invalid1/invalid2/invalid3"})) ;
+TEST_P(MeshExporter_test, checkBasicBehaviorInInvalidLongSubDirName)
+{
+    const TestParam& params = GetParam();
+    ASSERT_NO_THROW(this->checkBasicBehavior(params, tempdir+"/invalid1/invalid2/invalid3/outfile", {tempdir+"/invalid1/invalid2/invalid3"}));
 }
 
-TEST_P( MeshExporter_test, checkBasicBehaviorInInvalidRelativeDirName) {
-    std::vector<string> params = GetParam() ;
-    ASSERT_EQ(params.size(), NUM_PARAMS );
-    ASSERT_NO_THROW( this->checkBasicBehavior(params, "./invalidPath/outfile", {"./invalidPath"}) ) ;
+TEST_P(MeshExporter_test, checkBasicBehaviorInInvalidRelativeDirName)
+{
+    const TestParam& params = GetParam();
+    ASSERT_NO_THROW(this->checkBasicBehavior(params, "./invalidPath/outfile", {"./invalidPath"}));
 }
 
-TEST_P( MeshExporter_test, checkBasicBehaviorInValidDir) {
-    std::vector<string> params = GetParam() ;
-    ASSERT_EQ(params.size(), NUM_PARAMS );
-    ASSERT_NO_THROW(this->checkBasicBehavior(params, tempdir, {tempdir+"/exporter1."+params[0]}))  ;
+TEST_P(MeshExporter_test, checkBasicBehaviorInValidDir)
+{
+    const TestParam& params = GetParam();
+    constexpr std::string_view filename = "exporter1";
+    std::vector<std::string> paths { tempdir + "/" + std::string(filename) + "." + params.extension};
+    for (const auto& addExtension : params.additionalExtensions)
+    {
+        paths.push_back(tempdir + "/" + std::string(filename) + "." + addExtension);
+    }
+    ASSERT_NO_THROW(this->checkBasicBehavior(params, tempdir, paths));
 }
 
-TEST_P( MeshExporter_test, checkSimulationWriteEachNbStep) {
-    std::vector<string> params = GetParam() ;
-    ASSERT_EQ(params.size(), NUM_PARAMS );
-    ASSERT_NO_THROW(this->checkSimulationWriteEachNbStep(params, tempdir, {tempdir+"/exporterA00001."+params[0],
-                                                        tempdir+"/exporterA00002."+params[0],
-                                                        tempdir+"/exporterA00003."+params[0],
-                                                        tempdir+"/exporterA00004."+params[0]}, 20)) ;
+TEST_P(MeshExporter_test, checkSimulationWriteEachNbStep)
+{
+    const TestParam& params = GetParam();
+    constexpr std::string_view filename = "exporterA";
+    std::vector<std::string> paths;
+    for (unsigned int i = 0; i < 20; ++i)
+    {
+        std::stringstream ss;
+        ss << std::setw(5) << std::setfill('0') << i;
+
+        paths.push_back( tempdir + "/" + std::string(filename) + ss.str() + "." + params.extension);
+        for (const auto& addExtension : params.additionalExtensions)
+        {
+            paths.push_back(tempdir + "/" + std::string(filename)+ ss.str() + "." + addExtension);
```

Comment:
I don't get where the local variable `paths ` is used in this test?

Reply:
Yes, good catch. I'll fix it